src/app/home/work-item-widget/work-item-widget.component.ts

```diff
@@ -20,15 +20,19 @@ export class WorkItemWidgetComponent implements OnDestroy, OnInit  {
   loggedInUser: User;
   recentSpaces: Space[] = [];
   recentSpaceIndex: number = 0;
+  spaces: Space[] = [];
   subscriptions: Subscription[] = [];
   workItems: WorkItem[] = [];
 
   constructor(
+      private spaceService: SpaceService,
       private spacesService: Spaces,
       private workItemService: WorkItemService,
       private userService: UserService) {
-    this.subscriptions.push(userService.loggedInUser.subscribe(user => {
-      this.loggedInUser = user;
+    this.loggedInUser = this.userService.currentLoggedInUser;
+    this.subscriptions.push(this.spaceService.getSpacesByUser(this.loggedInUser.attributes.username).subscribe(spaces => {
+      this.spaces = spaces;
+      this.loading = false;
     }));
     this.subscriptions.push(spacesService.recent.subscribe(spaces => {
       this.recentSpaces = spaces;
```

Comment:
If we end up showing all user spaces, then we don't need to retrieve recent spaces. The fetchRecentSpace() method (not shown here) should be removed as well.

Reply:
@dlabrecq hm, I didn't realize there wasn't an easy way of retrieving all work-items owned by a particular user; I see the concern regarding many API calls.

What are your thoughts on treating this like the Profile work-items component, where the combobox is populated based on the results of `userService.getSpacesByUser` but the auto-population of work items is performed using `spacesService.recent`? This way there can be fewer API calls upfront to populate the widget with recent work items, but leave the functionality and freedom for the user to afterwards select any of their spaces to view work items?

Reply:
^ ping @dlabrecq

Reply:
At one time, the dashboard widget was labeled "Recent Work Items'. It allows me to view work items I recently created in the openshift.io space, for example. The profile page doesn't allow me to do that because I can only see my spaces.

If we only show the user's spaces, I cannot view my work items from the openshift.io space. The workaround we discussed with UXD was to show recent spaces, so users can potentially reach all their work items. That's what we have implemented today.

Changing the dashboard widget to show only the user's spaces is a question for @catrobson. The code for the two widgets is almost identical otherwise -- they both use ``spacesService.recent``.

Ideally, we would have an API that provides all of the user's work items, across all system spaces. I want to view ALL my work items on the dashboard as well as the profile page.

Reply:
@dlabrecq Ah that's a good point, I had forgotten about the case where you may have a WI in another user's space. In which case, there is an open issue regarding an API as you have described (https://github.com/openshiftio/openshift.io/issues/3760), but the collaborators API is being deprecated so any functionality would be put into the new API.

In which case, sorry to be going around in loops, but it appears that your original critique may be a valid substitute for the time being:
> The other widgets for this dashboard (i.e., "recent spaces" and "Recent Active Pipelines") are showing recent spaces, so this could also be addressed with a label change.